client/src/pages/Events/Events.tsx

```diff
@@ -5,27 +5,96 @@ import { useAuth } from "../../contexts/AuthContext";
 import { Navigate } from "react-router-dom";
 import { useFutureEvents } from "../eventHooks";
 import { toastNotify } from "../../utils/ui";
+import { processVolunteerCount } from "../ViewEvent/VolunteersTable";
+import { useEffect, useState } from "react";
+import { ProcessedScheduledSlot } from "../../types"; // Ensure this path is correct
 
 const newEventLink =
   "https://airtable.com/shrETAYONKTJMVTnZ?prefill_Supplier=Rap+4+Bronx&prefill_Start+Time=01/01/2023+09:00am&prefill_End+Time=01/01/2023+01:00pm&prefill_First+Driving+Slot+Start+Time=01/01/2023+10:30am&prefill_How+long+should+each+Driver+Time+Slot+be?=0:15&prefill_Max+Count+of+Drivers+Per+Slot=30&prefill_How+long+should+the+Logistics+slot+be?=1:30&prefill_Maximum+number+of+drivers+needed+for+this+event+(usually+30)?=30&prefill_Max+Count+of+Distributors+Per+Slot=30";
 
+//made to avoid implicit any type errors
+interface Event {
+  id: string;
+  dateDisplay: string;
+  time: string;
+  mainLocation: string;
+  numTotalParticipants: number;
+  numDrivers: number;
+  numPackers: number;
+  scheduledSlots: string[];
+}
+
+interface EventWithVolunteersData extends Event {
+  volunteers: ProcessedScheduledSlot[];
+}
+
 export function Events() {
   const { token, setToken } = useAuth();
   if (!token) {
     return <Navigate to="/" />;
   }
   const futureEventsQuery = useFutureEvents();
+  const [eventsData, setEventsData] = useState<EventWithVolunteersData[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [displayCount, setDisplayCount] = useState(8);
+
+  useEffect(() => {
+    const fetchVolunteersForEvent = async (event: Event, token: string) => {
+      const response = await fetch(
+        `/api/volunteers/?scheduledSlotsIds=${event.scheduledSlots.join(",")}`,
```

Comment:
The slowness is definitely coming from here.  The current app makes a single API call to /events endpoint and has all it needs to render the page for all future events. With this change, you've added calling the /volunteers endpoint once per event. So, if there are 50 future events, that's 51 total API calls instead of 1.  

So, the question is really what is the need for making an API call per event? I didn't anticipate that new API calls would be needed, and not per event. We should be pushing this to the backend and Airtable layer so that the data that is needed comes in the usual /events API call. So, it really should not be any additional API calls at all.  If we need to make a change to the Airtable table/view, that's preferable and expected, imo.

Reply:
@jadenar07, I added a new field to Supplier Pickup Events table called "numGuests". This is a rollup field in Airtable that calculates the number of guests attending the event. As such, you can modify the events.ts file to add that field to the response in the existing API call. Then you can work with that to populate the values in frontend. This should be a minor change only and probably makes sense to do as a new PR.

I've added the new field to both the prod and the development Airtable bases.